### PR TITLE
Shapeshifting inside of a pipe now violently kills you

### LINKED ISSUE #41289: Dio is disappointed
Okay, so, this Halloween, I learned something cool about vampires: if you shapeshift out of bat form while ventcrawling, you appear on the tile that covers wherever you happened to be when you shapeshifted out of bat form, no exit vent needed.

I also learned something uncool about vampires: If you perform that trick while you have a holoparasite in you, the holoparasite gets ghosted. I tested this with soulstones with a shade in them too, and sure enough, a shade that was walking (well, technically floating, I guess) and talking before I scooped it up and entered the vent went braindead after I exited the pipe by shapeshifting and dumped it out of the soulstone. Notably, this DIDN'T happen with my holoparasite when I exited ventcrawling "normally", nor when I just shapeshifting into and out of bat form.

### PR DESCRIPTION
**Pull request by tralezab**
fixes #41289

### CODE REVIEW

**Comment by tralezab** [comment]
let me test this before you merge it

**Comment by Thunder12345** [comment]
You get bonus points if blood comes out the nearest few vents

**Comment by SgtxRootbeer** [comment]
Can we like, warn mobs that this will kill?

**Comment by tralezab** [comment]
@Thunder12345 I like it, i'll see what I can do

@SgtxRootbeer I can compromise by checking if the animal you are shapeshifting into can ventcrawl, and blending you if it cannot. let me

**Comment by Supermichael777** [comment]
How about just not letting this happen instead of arbitrary punishments

**Comment by Barhandar** [comment]
Yeah, "ha ha fuck you you're permadead now" belongs in roguelikes, not in SS13.

**Comment by tralezab** [comment]
"ha ha fuck you you're permadead now" doesn't exist in SS13, but "did something really stupid with powerful wizard magic killed me" does. it completely works if what you're turning into can ventcrawl, you just have to be stupid to die with it.

I'll stick with my idea but lets see what the maintainer says

**Comment by creeperparty568** [comment]
> be me
> finally roll antag, nice
> in pipe
> want to shapeshift
> FUCKING GIBBED WITHOUT WARNING

File: code/modules/spells/spell_types/shapeshift.dm [hunk 1]
```diff
@@ -43,7 +43,16 @@
 			if(!shapeshift_type) //If you aren't gonna decide I am!
 				shapeshift_type = pick(animal_list)
 			shapeshift_type = animal_list[shapeshift_type]
-
+		
+		if(M.movement_type & (VENTCRAWLING)) //probably not a great idea after all
+			var/turf/turfyoudieon = get_turf(M)
+			var/obj/machinery/atmospherics/pipe/pipeyoudiein = locate() in turfyoudieon
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/spells/spell_types/shapeshift.dm]: test with multilayer pipes please...

File: code/modules/spells/spell_types/shapeshift.dm [hunk 2]
```diff
@@ -43,7 +43,16 @@
...
+			if(!turfyoudieon || !pipeyoudiein) //how are you ventcrawling if there is no pipe or turf
+				return
+			to_chat(M, "<span class='userdanger'>Shapeshifting inside of [pipeyoudiein] quickly turns you into a bloody mush.</span>")
+			M.remove_ventcrawl()
+			M.gib()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/spells/spell_types/shapeshift.dm]: move body out to the turf first so you can see the gibs!

**Comment by ShizCalev** [comment]
What @SgtxRootbeer said. Just preventing the transformation from taking place over randomly permanently killing a player is a far better solution than this.

**Pull request closed without merging**